zipkin-collector/core/src/main/java/zipkin2/collector/Collector.java

```diff
@@ -227,7 +228,13 @@ RuntimeException doError(String message, Throwable e) {
         message = format("%s due to %s(%s)", message, e.getClass().getSimpleName(), error);
         warn(message, e);
       }
-      return new RuntimeException(message, e);
```

Comment:
Very true.  Want me to update the conditional?

Reply:
here's example logic for here and otherwise

```java
    } catch (ExecutionException e) {
      Throwable cause = e.getCause();
      if (cause instanceof Error) throw (Error) cause;
      if (cause instanceof IOException) throw (IOException) cause;
      if (cause instanceof RuntimeException) throw (RuntimeException) cause;
      throw new RuntimeException("Unexpected on throttled call", cause);
    }
```

Reply:
Complications with this specific method:
1. We're expected to return a RuntimeException so I can't avoid wrapping Errors without specifically throwing them.
2. Part of the purpose of this method seems to be to wrap the Exception and add a specific message.  There are a lot of tests around this fact.  Example test failures:
    ```
    CollectorTest.errorDecoding_onErrorWithMessage:158 expected:<"[Cannot decode spans due to IllegalArgumentException(no beer)]"> but was:<"[no beer]">
    CollectorTest.errorDecoding_onErrorWithNullMessage:150 expected:<"Cannot decode spans due to RuntimeException()"> but was:<null>
    ```

I'd like to update this because I agree with the change but I feel like the change would be more worthy of a different issue/pull request.  What do you think?

Reply:
answering, but yeah this was bonus points it can happen in another PR. In fact I'll go ahead and do that now. Here's some notes.

1. errors are always allowed through, it is Throwable/Exception that can't get through.. the logic above was slightly modified from other code we use.
2. I think I wrote this code originally, there was no comment there.. it was not intended to wrap unnecessarily.

Reply:
ps by bonus points for fixing old logic I meant the old logic. new logic shouldn't be released unfixed... offer stands to fix on the way in.

Reply:
PS unraveling the old code is something I will do out-of-band. I think some should change, but yeah there is impact to error formatting, which some is done at the wrong place. Here only main thing is to not pass ExecutionException without unwrapping it properly. if that's done before formatting logic, should be ok